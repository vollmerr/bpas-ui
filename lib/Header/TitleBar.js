var _templateObject = _taggedTemplateLiteral(['\n  background: ', ';\n  height: ', 'px;\n'], ['\n  background: ', ';\n  height: ', 'px;\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n  min-width: ', 'px;\n'], ['\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n  min-width: ', 'px;\n']),
    _templateObject3 = _taggedTemplateLiteral(['\n  display: flex;\n'], ['\n  display: flex;\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../util/theme';

import Container from '../Container';
import TitleLogo from './TitleLogo';
import TitleText from './TitleText';
import TitleLinks from './TitleLinks';

var Wrapper = styled.div(_templateObject, theme.color.theme.primary, theme.size.titleBar);

var Content = styled.div(_templateObject2, theme.size.minWidth);

var Left = styled.div(_templateObject3);

/** Renders the headers title bar */
function TitleBar(_ref) {
  var title = _ref.title,
      _ref$isMobile = _ref.isMobile,
      isMobile = _ref$isMobile === undefined ? false : _ref$isMobile,
      onClick = _ref.onClick;

  return React.createElement(
    Wrapper,
    null,
    React.createElement(
      Container,
      null,
      React.createElement(
        Content,
        null,
        React.createElement(
          Left,
          null,
          React.createElement(TitleLogo, { onClick: onClick }),
          !isMobile && React.createElement(TitleText, { title: title })
        ),
        React.createElement(TitleLinks, { onClick: onClick })
      )
    )
  );
}

TitleBar.propTypes = {
  title: PropTypes.string.isRequired,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func
};

export default TitleBar;