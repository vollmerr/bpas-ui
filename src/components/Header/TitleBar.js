import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'react-bootstrap';
import styled from 'styled-components';
import theme from '../../theme';

import TitleText from './TitleText';
import TitleLinks from './TitleLinks';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  background: ${theme.color.primary};
  height: ${theme.size.titleBar}px;
`;

const Center = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

/** Renders the headers title bar */
function TitleBar({
  title,
  isMobile = false,
}) {
  return (
    <Wrapper>
      <Grid>
        <Center>
          <TitleText title={title} isMobile={isMobile} />
          <TitleLinks />
        </Center>
      </Grid>
    </Wrapper>
  );
}

TitleBar.propTypes = {
  title: PropTypes.string.isRequired,
  isMobile: PropTypes.bool,
};

export default TitleBar;
