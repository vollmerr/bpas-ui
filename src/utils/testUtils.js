import React from 'react';
import { StaticRouter } from 'react-router';

export function testRouter(component) {
  const context = {};
  const location = './';

  return (
    <StaticRouter location={location} context={context}>
      {component}
    </StaticRouter>
  );
}
