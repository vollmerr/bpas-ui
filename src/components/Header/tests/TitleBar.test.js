import React from 'react';
import renderer from 'react-test-renderer';
import { testRouter } from '../../../utils/testUtils.js';
import TitleBar from '../TitleBar';

const testTitle = 'test title';

describe('<TitleBar />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      testRouter(<TitleBar title={testTitle} />)
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('hides TitleText when isMobile', () => {});

  xit('passes the correct props down', () => {});
});
