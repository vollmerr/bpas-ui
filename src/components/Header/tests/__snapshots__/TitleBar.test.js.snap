// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleBar /> renders correctly 1`] = `
<div
  className="sc-hMqMXs emBvIB"
>
  <div
    className="container"
  >
    <div
      className="sc-kEYyzF bbYXyH"
    >
      <div
        className="sc-kkGfuU fESlcU"
      >
        <a
          href="http://www.ca.gov"
          onClick={undefined}
        >
          <img
            alt="CA Gov Logo"
            className="sc-ckVGcZ laVqmj"
            src="_caGovLogo.svg"
          />
        </a>
        <h1
          className="sc-jKJlTe jllqoO"
        >
          test title
        </h1>
      </div>
      <a
        className="sc-eNQAEJ cgLvGw"
        href="/help"
        onClick={[Function]}
      >
        Contact Us
      </a>
    </div>
  </div>
</div>
`;
